cylc/flow/task_proxy.py

```diff
@@ -178,6 +179,7 @@ class TaskProxy:
         'tokens',
         'try_timers',
         'waiting_on_job_prep',
+        'mode_settings',
```

Comment:
Make sure you preserve this variable in the `copy_to_reload_successor` method, otherwise it will get wiped on reload.

Reply:
Yes - That's a key thing that will make the code simpler.